### DISCUSSION TITLE
How Do I Record HLTV Events ?

### DISCUSSION THREAD

**Discussion by aniruddhapdeshpande99** [saul/demofile#314] [Questions] [answered]
I was looking to track HLTV events in a game, especially the Events wherein you can track when an Observer changes the POV to another player or is using a fixed camera on an elevated position (For Eg. To Display some utility usage)

I found this link here below that has a list of HLTV Events for your reference:
https://github.com/SteamDatabase/GameTracking-CSGO/blob/master/csgo/pak01_dir/resource/hltvevents.res

Thanks.

**Comment by saul** [comment, marked as answer]
Unfortunately the only HLTV events that are recorded in demo files are `hltv_status` and `hltv_chat`. You can get the camera man with the following code:

```ts
function getCameraMan(): Player | null {
  const cameraManAccountId = parseInt(
    demoFile.conVars.vars.get("spec_autodirector_cameraman") || "0"
  );
  if (isNaN(cameraManAccountId) || cameraManAccountId <= 0) return null;

  return demoFile.entities.getByAccountId(cameraManAccountId);
}
```

However I don't think this `spec_autodirector_cameraman` convar is used. The game client does include logic for an 'auto director' that swaps between players/actions that are interesting, but this is something that you'd have to reimplement yourself.
